### ISSUE TITLE
AssertionError in KeyInfo.__str__

### ISSUE THREAD

**Issue by The-Compiler** [qutebrowser/qutebrowser#3672]
Reported by @forkbong at https://crashes.qutebrowser.org/view/c85d6e35

> pressed alt-shift, which is configured to change the keyboard layout (not bound to anything on qutebrowser)

```
12:59:15 DEBUG    keyboard   basekeyparser:_debug_log:89 Got key: 0x1000020 / text: '<Shift>' / dry_run True
12:59:15 DEBUG    keyboard   basekeyparser:_debug_log:89 Ignoring, only modifier
12:59:15 DEBUG    modes      modeman:_handle_keypress:187 match: 0, forward_unbound_keys: auto, passthrough: False, is_non_alnum: True, dry_run: True --> filter: False (focused: <PyQt5.QtWidgets.QWidget object at 0x7fa9800eb3a8>)
12:59:15 DEBUG    modes      modeman:_handle_keypress:160 got keypress in mode KeyMode.normal - delegating to <qutebrowser.keyinput.modeparsers.NormalKeyParser>
12:59:15 DEBUG    keyboard   basekeyparser:_debug_log:89 Got key: 0x1000020 / text: '<Shift>' / dry_run True
12:59:15 DEBUG    keyboard   basekeyparser:_debug_log:89 Ignoring, only modifier
12:59:15 DEBUG    modes      modeman:_handle_keypress:187 match: 0, forward_unbound_keys: auto, passthrough: False, is_non_alnum: True, dry_run: True --> filter: False (focused: <PyQt5.QtWidgets.QWidget object at 0x7fa9800eb3a8>)
12:59:15 DEBUG    modes      modeman:_handle_keypress:160 got keypress in mode KeyMode.normal - delegating to <qutebrowser.keyinput.modeparsers.NormalKeyParser>
12:59:15 DEBUG    keyboard   basekeyparser:_debug_log:89 Got key: 0x1000020 / text: '<Shift>' / dry_run False
12:59:15 DEBUG    keyboard   basekeyparser:_debug_log:89 Ignoring, only modifier
12:59:15 DEBUG    modes      modeman:_handle_keypress:187 match: 0, forward_unbound_keys: auto, passthrough: False, is_non_alnum: True, dry_run: False --> filter: False (focused: <PyQt5.QtWidgets.QWidget object at 0x7fa9800eb3a8>)
12:59:15 DEBUG    modes      modeman:_handle_keypress:160 got keypress in mode KeyMode.normal - delegating to <qutebrowser.keyinput.modeparsers.NormalKeyParser>
12:59:15 ERROR    misc       crashsignal:exception_hook:216 Uncaught exception
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/qutebrowser/app.py", line 933, in eventFilter
    return handler(event)
  File "/usr/lib/python3.6/site-packages/qutebrowser/app.py", line 906, in _handle_key_event
    return man.handle_event(event)
  File "/usr/lib/python3.6/site-packages/qutebrowser/keyinput/modeman.py", line 328, in handle_event
    return handler(event)
  File "/usr/lib/python3.6/site-packages/qutebrowser/keyinput/modeman.py", line 161, in _handle_keypress
    match = parser.handle(event, dry_run=dry_run)
  File "/usr/lib/python3.6/site-packages/qutebrowser/keyinput/modeparsers.py", line 79, in handle
    match = super().handle(e, dry_run=dry_run)
  File "/usr/lib/python3.6/site-packages/qutebrowser/keyinput/basekeyparser.py", line 133, in handle
    txt = str(keyutils.KeyInfo.from_event(e))
  File "/usr/lib/python3.6/site-packages/qutebrowser/keyinput/keyutils.py", line 234, in __str__
    assert len(key_string) == 1, key_string
AssertionError
```

**Comment by The-Compiler** [comment]
Updated report: https://crashes.qutebrowser.org/view/a5885aed

Looks like the key is 0x0.

> I change the keyboard layout using `setxkbmap -layout us,gr -option grp:alt_shift_toggle,grp_led:caps,ctrl:nocaps`
> If it's not configured like that, it doesn't crash

**Comment by The-Compiler** [comment]
Fixed in d1854eddaffe66c4bbeb2e819a57dc8dbfc6d703.
